src/commonMain/kotlin/software/momento/kotlin/sdk/exceptions/CacheServiceExceptionMapper.kt

```diff
@@ -24,36 +25,47 @@ public object CacheServiceExceptionMapper {
         return when (e) {
             is IllegalArgumentException -> InvalidArgumentException(e.message)
             is SdkException -> e
+            is StatusException -> {
+                val statusCode = e.status.code
+                val errorDetails = MomentoTransportErrorDetails(
+                    MomentoGrpcErrorDetails(statusCode, e.message!!, metadata)
+                )
+                convertStatusException(e, errorDetails, statusCode)
+            }
             is StatusRuntimeException -> {
                 val statusCode = e.status.code
                 val errorDetails = MomentoTransportErrorDetails(
                     MomentoGrpcErrorDetails(statusCode, e.message!!, metadata)
                 )
-                when (statusCode) {
-                    Status.Code.INVALID_ARGUMENT, Status.Code.UNIMPLEMENTED,
-                    Status.Code.OUT_OF_RANGE, Status.Code.FAILED_PRECONDITION -> BadRequestException(
-                        e, errorDetails
-                    )
-                    Status.Code.CANCELLED -> CancellationException(e, errorDetails)
-                    Status.Code.DEADLINE_EXCEEDED -> TimeoutException(e, errorDetails)
-                    Status.Code.PERMISSION_DENIED -> PermissionDeniedException(
-                        e, errorDetails
-                    )
-                    Status.Code.UNAUTHENTICATED -> AuthenticationException(e, errorDetails)
-                    Status.Code.RESOURCE_EXHAUSTED -> LimitExceededException(
-                        e, errorDetails
-                    )
-                    Status.Code.NOT_FOUND -> NotFoundException(e, errorDetails)
-                    Status.Code.ALREADY_EXISTS -> AlreadyExistsException(e, errorDetails)
-                    Status.Code.UNKNOWN -> UnknownServiceException(e, errorDetails)
-                    Status.Code.UNAVAILABLE -> ServerUnavailableException(e, errorDetails)
-                    Status.Code.ABORTED, Status.Code.INTERNAL, Status.Code.DATA_LOSS -> InternalServerException(
-                        e, errorDetails
-                    )
-                    else -> InternalServerException(e, errorDetails)
-                }
+                convertStatusException(e, errorDetails, statusCode)
```

Comment:
nit the `StatusRuntimeException` block looks identical to the `StatusException` block. I think kotlin lets you do 

```
is StatusException, StatusRuntimeException -> { // the code block }

Reply:
I didn't combine them because they don't have a common parent. I can clean it up a bit with a nested `when` to get the status code out of the exception. I'll do that.